include/oneapi/dpl/internal/dynamic_selection_impl/round_robin_policy.h

```diff
@@ -0,0 +1,123 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_ROUND_ROBIN_POLICY_IMPL_H
+#define _ONEDPL_ROUND_ROBIN_POLICY_IMPL_H
+
+#include <atomic>
+#include "oneapi/dpl/internal/dynamic_selection_impl/scoring_policy_defs.h"
+
+namespace oneapi {
+namespace dpl{
+namespace experimental{
+
+  template <typename Backend>
+  struct round_robin_policy {
+  private:
+    using resource_container_t = typename Backend::resource_container_t;
+    using resource_container_size_t = typename resource_container_t::size_type;
+  public:
+    using resource_t = typename Backend::resource_t; 
+    using selection_t = oneapi::dpl::experimental::basic_selection_t<resource_t>; 
+    using wait_t = typename Backend::wait_t; 
+    using submission_group_t = Backend::submission_group_t; 
+
+  private:
+    std::shared_ptr<Backend> backend_;
+
+    struct unit_t{
+        resource_container_t resource_;
+        resource_container_size_t num_contexts_;
+        std::atomic<resource_container_size_t> next_context_;
+	int offset_;
+	bool deferred_;
+    };
+
+    std::shared_ptr<unit_t> unit_;
+  public:
+    round_robin_policy(int offset=0, bool deferred=false) : backend_{std::make_shared<Backend>()}, unit_{std::make_shared<unit_t>()}  {
```

Comment:
Should be a pre-defined deferred value for the offset, not an additional argument. And if deferred nothing should be initialized in policy. Since backends do require an initialize function, it should be ok to assume they can also be constructed as deferred. So it would be ok to create backend in deferred state.